bin/setup.sh

```diff
@@ -1,4 +1,19 @@
 #!/bin/sh
 set -e
 
-for PLUG in %PLUGS%; do sudo snap connect %SNAP%:${PLUG}; done
+snap_connect() {
+  if snap connections | grep --quiet "$1:$2"; then
+    return 0
+  fi
```

Comment:
This really makes the remaining changes in this file redundant, as `wayland` gets autoconnected to `ubuntu-frame` regardless of the order of installation?

Reply:
Wayland only gets auto connected if there is a single provider. And when developing snaps it isn't unusual to have two. (E.g. snapd and ubuntu-frame on a non-Core system.)

On desktop we need to connect wayland to snapd, on core to ubunut-frame or mir-kiosk, so that's why there's logic looking for slot provider. The previous code simply assumed snapd.